lib/Terminal/LineEditor/RawTerminalInput.rakumod

```diff
@@ -565,20 +565,9 @@ role Terminal::LineEditor::RawTerminalUtils {
     #| Detect terminal size, returning a Promise that will be kept with
     #| (rows, cols) or Empty if unable
     method detect-terminal-size() {
-        # XXXX: This query has been found to have compatibility problems; it
-        #       works in xterm and libvte terminals, but not in many others
-        #       such as Konsole, Windows Console, linux, etc.
-        # my $response = self.query-terminal("\e[18t", 't');
-        # $response ~~ /^ "\e[8;" (\d+) ';' (\d+) 't' $/ ?? (+$0, +$1) !! Empty
-
-        # Instead of the above, take advantage of clipping behavior of cursor
-        # movement commands by saving the cursor, requesting a move way outside
-        # the terminal, detecting the cursor position, and restoring the cursor.
-
-        $.output.print("\e7\e[9999;9999H");
-        my $p := self.detect-cursor-pos;
-        $.output.print("\e8");
-        $p
+        my $fd = $.output.native-descriptor;
+        my $size = Terminal::API::get-window-size($fd);
+        Promise.kept(($size.rows, $size.cols))
```

Comment:
Curious about the reason for this change ... does the clipping behavior trick not work on Windows?

Reply:
I didn't try, but would suspect that it'd work. I seem to recall that you explicitly asked for an API to retrieve the window size, I think the cause was Terminal-Print, where the size was retrieved via `tput cols/rows`. The change here was just for uniformness / simplicity. Would you prefer to keep it as it was before?

Reply:
Let's separate out this bit into a separate PR.  We may need it, but I'd rather not make this change as part of this PR.  (Though both small, the two diff blocks are in my mind two separate conceptual changes, one of which may or may not be needed.)